SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialConnectionManagerTest.cpp

```diff
@@ -0,0 +1,293 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+#include "Interop/Connection/SpatialConnectionManager.h"
+#include "CoreMinimal.h"
+
+#define CONNECTIONMANAGER_TEST(TestName) \
+	GDK_TEST(Core, SpatialConnectionManager, TestName)
+
+class FTemporaryCommandLine
+{
+public:
+    FTemporaryCommandLine(const FString& NewCommandLine)
+    {
+    	OldCommandLine = FCommandLine::GetOriginal();
+    	FCommandLine::Set(*NewCommandLine);
+    }
+
+	~FTemporaryCommandLine()
+    {
+    	FCommandLine::Set(*OldCommandLine);
+    }
+
+private:
+    FString OldCommandLine;
+};
+
+CONNECTIONMANAGER_TEST(SetupFromURL_Locator_CustomLocator)
```

Comment:
We usually format the test name in a certain way:

An example:

"GIVEN_a_static_component_view_with_no_data_WHEN_asking_load_balance_enforcer_for_acl_assignments_THEN_return_no_acl_assignment_requests"

We have a "GIVEN", a "WHEN" and a  "THEN". 

"GIVEN" - the starting state of the test
"WHEN" - the tested functionality
"THEN" - what should happen

Reply:
This is probably contentious. I've seen that style in the other tests and started out the same way. However, while there is a certain merit in giving the tests very explicit names, it makes finding a specific test in a list very hard. Compare the test names in this picture, for example. 
![image](https://user-images.githubusercontent.com/52573875/81056658-cb125f80-8ec2-11ea-9791-32247839c93c.png)

I can't make out any commonalities or differences in the lower test names with a quick glance. I have to concentrate and read every single one of them.
In contrast, I can quickly group tests into categories for the upper ones and find the one I'm interested in. Personally, I also don't find it hard to understand what they do, but since I wrote  and know them, I might be biased.

What do you think?

Reply:
I also agree that it does make things difficult to read in the UI you presented. I think possibly what we should do in the GDK is to comment each test with a similar format to the "GIVEN, WHEN, THEN" titles we use currently. We should then use easy to read names for the actual test name so that it makes parsing the UI easier. I would bring this up in the safespace channel and see what peoples opinions are on test naming conventions.

Reply:
Since no one seems to complain against changing the test names, I suppose we can use this PR as the first example